components/integrations/bots/bot.jsx

```diff
@@ -0,0 +1,488 @@
+                                        defaultMessage='Token Description: '
+                                    />
+                                </label>
+                                <div className='col-sm-4'>
+                                    <input
+                                        autoFocus={true}
+                                        className='form-control form-sm'
+                                        type='text'
+                                        maxLength={64}
+                                        value={this.state.token.description}
+                                        onChange={this.handleUpdateDescription}
+                                    />
+                                </div>
+                            </div>
+                            <div>
+                                <div className='padding-top padding-bottom'>
+                                    <FormattedMessage
+                                        id='user.settings.tokens.nameHelp'
+                                        defaultMessage='Enter a description for your token to remember what it does.'
+                                    />
+                                </div>
+                                <label
+                                    id='clientError'
+                                    className='has-error is-empty'
+                                >
+                                    {this.state.error}
+                                </label>
+                                <div className='margin-top'>
+                                    <SaveButton
+                                        btnClass='btn-sm btn-primary'
+                                        savingMessage={
+                                            <FormattedMessage
+                                                id='user.settings.tokens.save'
+                                                defaultMessage='Save'
+                                            />
+                                        }
+                                        saving={false}
+                                    />
+                                    <button
+                                        className='btn btn-sm btn-link'
+                                        onClick={this.closeCreateToken}
+                                    >
+                                        <FormattedMessage
+                                            id='user.settings.tokens.cancel'
+                                            defaultMessage='Cancel'
+                                        />
+                                    </button>
+                                </div>
+                            </div>
+                        </form>
+                    </div>
+                </div>
+            );
+        } else if (this.state.creatingTokenState === 'CREATED') {
+            tokenList.push(
```

Comment:
This component is big.  Maybe move other into a separate component like those item being pushed into `tokenList`?

Reply:
That makes sense. Will make a ticket to do that as part of the improvements to this UI before release in 5.12.